tests/unit/inventory/test_graphql.py

```diff
@@ -0,0 +1,98 @@
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+import pytest
+import json
+import os
+
+from functools import partial
+from unittest.mock import patch, MagicMock, Mock, call
+from ansible.inventory.data import InventoryData
+from ansible.errors import AnsibleError
+from ansible.utils.display import Display
+
+try:
+    from ansible_collections.networktocode.nautobot.plugins.inventory.gql_inventory import InventoryModule
+
+except ImportError:
+    import sys
+
+    # Not installed as a collection
+    # Try importing relative to root directory of this ansible_modules project
+
+    sys.path.append("plugins/inventory/")
+    sys.path.append("tests")
+    from gql_inventory import InventoryModule
+
+
+def load_graphql_device_data(path, test_path):
+    with open(f"{path}/test_data/{test_path}/device_data.json", "r") as f:
+        data = json.loads(f.read())
+    return data
+
+
+load_relative_test_data = partial(load_graphql_device_data, os.path.dirname(os.path.abspath(__file__)))
+
+
+@pytest.fixture
+def inventory_fixture():
+    inventory = InventoryModule()
+    inventory.inventory = InventoryData()
+    inventory.inventory.add_host("mydevice")
+
+    return inventory
+
+
+@pytest.fixture
+def device_data():
+    json_data = load_relative_test_data("graphql_groups")
+    return json_data
+
+
```

Comment:
Could you add some tests for the new functions here as well? Just to validate it's working as intended for the Ansible host and Ansible Network Host variables.

Reply:
Added!